assets/js/common/Table/search.js

```diff
@@ -0,0 +1,23 @@
+const canonicalise = (val) => val.normalize().trim().toLowerCase();
+
+const stringContains = (subject, predicate) =>
+  canonicalise(subject).includes(canonicalise(predicate));
+
+const arrayContains = (subject, predicate) =>
+  subject
+    .map((it) => stringContains(it, predicate))
+    .reduce((accumulator, currentValue) => accumulator || currentValue, false);
+
+const contains = (subject, predicate) =>
```

Comment:
I think you could cast everything to a string array and search over that. But as long as the API does not change, it can always be adjusted, when needed

Reply:
If we move this search feature outside of `Table`, then I think this problem mostly goes away. Because you will know the shape of the specific data you are dealing with, you no longer have to account for every type